File: app-tracker-automation/pages/login_page.py
"""
Login Page Object Model
Handles login functionality for the insurance portal
"""

from playwright.sync_api import Page, expect
from pages.base_page import BasePage
from locators.login_locators import LoginLocators


class LoginPage(BasePage):
    """Login page object"""
    
    def __init__(self, page: Page):
        super().__init__(page)
        self.locators = LoginLocators()
    
    def load(self):
        """Load login page"""
        self.navigate_to("/login")
        self.wait_for_page_load()
    
    def login(self, username: str, password: str):
        """Perform login with credentials"""
        self.logger.info(f"Attempting login for user: {username}")
        
        # Enter username
        self.page.locator(self.locators.username_input).fill(username)
        
        # Enter password
        self.page.locator(self.locators.password_input).fill(password)
        
        # Click login button
        self.page.locator(self.locators.login_button).click()
        
        # Wait for login to complete
        self.waits.wait_for_element_disappear(self.locators.login_button)
    
    def is_login_page_displayed(self) -> bool:
        """Check if login page is displayed"""
        return self.is_element_visible(self.locators.login_form)
    
    def get_error_message(self) -> str:
        """Get login error message"""
        return self.page.locator(self.locators.error_message).text_content()
    
    def is_login_successful(self) -> bool:
        """Check if login was successful"""
        return not self.is_element_visible(self.locators.login_form)

File: ts-framework/pages/login_page.ts
import { Page } from '@playwright/test';
import { BasePage } from './base_page';
import { LoginLocators } from '../locators/login_locators';

export class LoginPage extends BasePage {
  private locators: LoginLocators;

  constructor(page: Page) {
    super(page);
    this.locators = new LoginLocators();
  }

  async load(): Promise<void> {
    await this.navigateTo('/login');
    await this.waitForPageLoad();
  }

  async login(username: string, password: string): Promise<void> {
    this.logger.info(`Attempting login for user: ${username}`);

    // Enter username
    await this.page.locator(this.locators.username_input).fill(username);

    // Enter password
    await this.page.locator(this.locators.password_input).fill(password);

    // Click login button
    await this.page.locator(this.locators.login_button).click();

    // Wait for login to complete
    await this.waits.waitForElementDisappear(this.locators.login_button);
  }

  async isLoginPageDisplayed(): Promise<boolean> {
    return await this.isElementVisible(this.locators.login_form);
  }

  async getErrorMessage(): Promise<string> {
    return await this.page.locator(this.locators.error_message).textContent() || '';
  }

  async isLoginSuccessful(): Promise<boolean> {
    return !(await this.isElementVisible(this.locators.login_form));
  }
}